File: app/src/main/java/org/doomer/qnotez/utils/ActivityUtils.java
package org.doomer.qnotez.utils;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;

import org.doomer.qnotez.R;
import org.doomer.qnotez.fragments.MainFragment;
import org.doomer.qnotez.fragments.TrashFragment;


public class ActivityUtils {
    public static final int BACK_SAVE_NO = 0;
    public static final int BACK_SAVE_CONFIRM = 1;
    public static final int BACK_SAVE_AUTO = 2;

    public static int noteEditBackBehavior(Context ctx) {
        SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(ctx);
        String key = ctx.getString(R.string.settings_key_back_action_note_edit);

        return Integer.parseInt(sp.getString(key, String.valueOf(BACK_SAVE_CONFIRM)));
    }

    public static void changeFragment(AppCompatActivity act, int layoutId, String tag) {

        Fragment frg = null;

        if (tag.equals(TrashFragment.FRAGMENT_TAG)) {
            frg = new TrashFragment();
        } else {
            frg = new MainFragment();
        }

        if (frg != null) {
            act.getSupportFragmentManager().beginTransaction()
                    .replace(layoutId, frg, tag)
                    .commit();
        }
    }

}

File: app/src/main/kotlin/org/doomer/qnotez/utils/ActivityUtils.kt
package org.doomer.qnotez.utils

import android.app.Activity
import android.content.Context
import android.preference.PreferenceManager
import android.support.design.widget.Snackbar
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Toast

import org.doomer.qnotez.R
import org.doomer.qnotez.ui.fragments.MainFragment
import org.doomer.qnotez.ui.fragments.TrashFragment


object ActivityUtils {
    val BACK_SAVE_NO = 0
    val BACK_SAVE_CONFIRM = 1
    val BACK_SAVE_AUTO = 2

    fun noteEditBackBehavior(ctx: Context): Int {
        val sp = PreferenceManager.getDefaultSharedPreferences(ctx)
        val key = ctx.getString(R.string.settings_key_back_action_note_edit)

        return Integer.parseInt(sp.getString(key, BACK_SAVE_CONFIRM.toString()))
    }

    // DEPRECATED on 3.0
    // DELETED on 3.x
    // TODO - change it on androidx.arch.navigation
    fun changeFragment(act: AppCompatActivity, layoutId: Int, tag: String) {

        var frg: Fragment? = null

        if (tag == TrashFragment.FRAGMENT_TAG) {
            frg = TrashFragment()
        } else {
            frg = MainFragment()
        }

        if (frg != null) {
            act.supportFragmentManager.beginTransaction()
                    .replace(layoutId, frg, tag)
                    .commit()
        }
    }
}

fun Activity.showMessageSnack(view : View, msg : String) {
    Snackbar.make(view, msg, Snackbar.LENGTH_LONG).show()
}

fun Activity.showMessageToast(msg : String) {
    Toast.makeText(this, msg, Toast.LENGTH_LONG).show()
}


